js/zap-common.js

```diff
@@ -39,32 +34,42 @@ function getYouTubeVideoDescription(videoID, youTubeDataApiKey, onSuccess, onFai
     }).fail(onFail);
 }
 
-// The url parameter could be the video ID
-function parseYoutubeVideoID(url) {
+/**
+ * Given a YouTube video ID, YouTube.com video URL, or YouTu.Be video URL, return an object
+ * containing the method used to derive the ID, and the ID itself. The following were used to test;
+ * FdeioVndUhs
+ * https://youtu.be/FdeioVndUhs
+ * https://www.youtube.com/watch?v=FdeioVndUhs&feature=youtu.be
+ *
+ * @param {string} uri - YouTube video ID, YouTube.com url, or YouTu.be url.
+ * @returns {{format: string, id: string}} - Object containing format and YouTube ID.
+ */
+function parseYoutubeVideoID(uri) {
     var videoInfo = {
         format: "other",
         id: null
     };
     var shortUrlDomain = "youtu.be";
     var longUrlDomain = "youtube.com";
 
-    if (url && url.length > 0) {
+    var url = URI(uri);
+    var domain = url.domain();
+
+    if (uri && uri.length > 0) {
         // youtube.com format
-        if (url.indexOf(longUrlDomain) !== -1) {
+        if (domain === longUrlDomain) {
             videoInfo.format = longUrlDomain;
-            videoInfo.id = getParameterByName(url, "v");
+            videoInfo.id = url.search(true).v;
         }
         // youtu.be format
-        else if (url.indexOf(shortUrlDomain) !== -1) {
+        else if (domain === shortUrlDomain) {
             videoInfo.format = shortUrlDomain;
-            var endPosition = url.indexOf("?") === -1 ? url.length : url.indexOf("?");
-            var offset = url.indexOf(shortUrlDomain) + shortUrlDomain.length + 1; // Skip over the slash also
-            videoInfo.id = url.substring(offset, endPosition);
+            videoInfo.id = url.path().slice(1);
         }
         // Assume YouTube video ID string
         else {
             videoInfo.format = "video ID";
-            videoInfo.id = url;
+            videoInfo.id = uri;
         }
 
         var slashPos = videoInfo.id.indexOf("/");
```

Comment:
Does this case get handled by the URI library? If so, we can remove these 4 lines